File: 4030-check-ascii-palindromic/4030-check-ascii-palindromic.java
class Solution {
    public boolean isPalindromic(String s) {
        String output = "";
        for(int i=0; i<s.length(); i++){
            output += intToBin(s.charAt(i)-0);
        }
        int l=0, r=output.length()-1;
        while(l<=r){
            if(output.charAt(l) != output.charAt(r))
                return false;
            l++;
            r--;
        }
        return true;
    }
    String intToBin(int num){
        String ans = "";
        for(int i=7; i>=0; i--){
            if((num & (1<<i)) != 0){
                ans += '1';
            }
            else{
                ans += '0';
            }
        }
        return ans;
    }
}

File: 4030-check-ascii-palindromic/4030-check-ascii-palindromic.cpp
class Solution {
public:
    string intToBin(int num){
        string ans = "";
        for(int i=7; i>=0; i--){
            if(num & (1<<i)){
                ans += '1';
            }
            else{
                ans += '0';
            }
        }
        return ans;
    }
    bool isPalindromic(string s) {
        string output = "";
        for(int i=0; i<s.size(); i++){
            output += intToBin(s[i]-0);
        }
        int l=0, r=output.size()-1;
        while(l<=r){
            if(output[l] != output[r])
                return false;
            l++;
            r--;
        }
        return true;
    }
};